### ISSUE TITLE
`unicorn/prefer-combined-guards` reduces readability

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#3702]
Just a nit: I kinda like the rule in principle, but in practice I think it condenses the code with doubtful upsides. 

Like with `no-immediate-mutation` (https://github.com/sindresorhus/eslint-plugin-unicorn/issues/3700), the rule looks good only on paper with short `foo && bar` examples, less so with real-world ones.

Example:

```js
// ❌⁉️
// Exclude explicitly linked comments #5363
if (commentText === linkedComment) {
	continue;
}

if (!isLowQualityComment(commentText.textContent) && isFocused(comment)) {
	continue;
}

// Comments that contain useful images or links shouldn't be removed
// Images are wrapped in <a> tags on GitHub hence included in the selector
if (elementExists('a', commentText)) {
	continue;
}
```

Every condition is documented individually where needed, the code flows lightly and is readable. This rule pushes me to:

```js
// ✅⁉️
if (
	// Exclude explicitly linked comments #5363
	commentText === linkedComment

	|| (!isLowQualityComment(commentText.textContent) && isFocused(comment))

	// Comments that contain useful images or links shouldn't be removed
	// Images are wrapped in <a> tags on GitHub hence included in the selector
	|| elementExists('a', commentText)
	) {
	continue;
}
```

which requires extra operators and parens.

**Comment by fregante** [comment]
One more example where an already-merged chunk of conditions turns into a nested/parenthesized mega-condition:

```js
// ❌⁉️
if (!commit) {
	return;
}

if (
	// Skip commits that are _only_ "ci:" without anything else. Rare but it would be confusing to show just the label
	commit.raw === textNode.textContent
	&& !commitTitleElement.nextElementSibling

	// Ensure that the element contains only plain text, not stuff like <code>
	&& commitTitleElement.childElementCount < 1
) {
	return;
}
```



```js
// ✅⁉️
if (
	!commit
	|| (
		// Skip commits that are _only_ "ci:" without anything else. Rare but it would be confusing to show just the label
		commit.raw === textNode.textContent
		&& !commitTitleElement.nextElementSibling
		// Ensure that the element contains only plain text, not stuff like <code>
		&& commitTitleElement.childElementCount < 1
	)
) {
	return;
}
```


all of this because to avoid a "repeated" `if (!commit) {return}` ?
